server/services/interventionsService.ts

```diff
@@ -60,13 +60,17 @@ export default class InterventionsService {
     })) as DraftReferral
   }
 
-  async createDraftReferral(token: string): Promise<DraftReferral> {
+  async createDraftReferral(token: string, initialFields?: Partial<DraftReferral>): Promise<DraftReferral> {
     const restClient = await this.createRestClient(token)
 
-    return (await restClient.post({
+    const referral = (await restClient.post({
       path: `/draft-referral`,
       headers: { Accept: 'application/json' },
     })) as DraftReferral
+
+    return initialFields === undefined || initialFields === {}
```

Comment:
I reckon that if is just a temporary thing then we can just stick this in the controller action - I don't think we need this functionality in the service, and if we do then we should probably test it.

Reply:
i think the content is temporary, but i can see this being used in the future when we are starting the referral journey with a known crn and service etc.

what do you think ?

i agree on the testing part though. i'll add something.

Reply:
I suppose, but I kinda find it cleaner to stick with the current model of one `InterventionsService` method = one endpoint. Cos Pact tests are designed for testing a single request / response pair so we could no longer test this method with a Pact test. And it would be nice if we could keep all the tests for this class as Pact tests.

Reply:
(Also, if we wanted to allow passing initial data, could we actually make that part of the `POST` body?)

Reply:
i actually initially wrote this that way. but i decided that changing the API was unnessecary since it's possible with a POST then PATCH. furthermore it makes error handling much harder. e.g. if you send the POST with invalid initial data does the referral actually get created?

Reply:
Makes sense. I reckon the same applies here, then - there's no way of communicating to the caller of this method whether the referral got created and hence whether they need to try and recover by trying the PATCH again or by creating another referral.

Reply:
on the testing - i agree it makes the pact test not fully test the methods. but at the same time, limiting the logic we can have in the service layer in this way isn't ideal. i can write a normal unit test for this functionality aside from the contract tests. having the ability for a service level method to make multiple calls to achieve an 'action' like 'getOrCreate' for example, or 'createAndSet' (as we have here), or whatever, is good IMO.